src/main/scala/diplomacy/Nodes.scala

```diff
@@ -138,80 +344,296 @@ trait FormatNode[I <: FormatEdge, O <: FormatEdge] extends BaseNode {
   }
 }
 
+/**
+  * A Handle with no explicitly defined binding functionality.
+  *
+  * A [[NoHandle]] is at the top of the Handle type hierarchy, but it does not define
+  * any binding operators, so by itself a [[NoHandle]] cannot be used on either side
+  * of a bind operator.
```

Comment:
For example, a source node connected directly to a sink node produces a `NoHandle`, because there are no further bindings that could be applied to either side of the pair of nodes.